### PR TITLE
CLN: avoid accessing private functions

### PR DESCRIPTION
**Pull request by jbrockmendel**
xref #33394 

@simonjayhawkins thoughts on how to handle the mypy complaint?

### CODE REVIEW

**Comment by WillAyd** [comment]
Yea the recursive call obviously makes things difficult. I think this is a use case for overload, where the type of input dictates the return type

File: pandas/core/series.py [hunk 1]
```diff
@@ -2623,22 +2624,57 @@ def _binop(self, other, func, level=None, fill_value=None):
...
 
         with np.errstate(all="ignore"):
             result = func(this_vals, other_vals)
 
         name = ops.get_op_result_name(self, other)
-        ret = ops._construct_result(self, result, new_index, name)
+        ret = this._construct_result(result, name)
         return ret
 
+    def _construct_result(
```
Comment (jreback) [inline pandas/core/series.py:2641]: maybe contrust_result_from_op

Comment (jbrockmendel) [reply pandas/core/series.py:2641]: for now mirroring the naming in DataFrame

**Comment by jbrockmendel** [comment]
updated to fix the mypy concern

**Review by WillAyd — APPROVED** [review]
lgtm

**Pull request merged**